File: 0101-symmetric-tree/0101-symmetric-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
bool sol(TreeNode* left,TreeNode*right){
    if(!left &&!right){return true;}
    else if(!left &&right){return false;}
    else if(!right && left ){return false;}
    if(left->val==right->val){
        bool s1=sol(left->left,right->right);
        bool s2=sol(left->right,right->left);
        return s1&&s2;
    }
    return false;
}
    bool isSymmetric(TreeNode* root) {
        if(!root)return true;
        return sol(root->left,root->right);
    }
};

File: 0101-symmetric-tree/0101-symmetric-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    private boolean sol(TreeNode left, TreeNode right) {
        if (left == null && right == null) {
            return true;
        } else if (left == null || right == null) {
            return false;
        }
        if (left.val == right.val) {
            boolean s1 = sol(left.left, right.right);
            boolean s2 = sol(left.right, right.left);
            return s1 && s2;
        }
        return false;
    }

    public boolean isSymmetric(TreeNode root) {
        if (root == null) {
            return true;
        }
        return sol(root.left, root.right);
    }
}